packages/svelte/src/compiler/phases/2-analyze/css/Selector.js

```diff
@@ -836,6 +841,7 @@ function group_selectors(selector) {
 			block = new Block(child);
 			blocks.push(block);
 		} else {
+			if (child.type === "NestedSelector") block.nested = true;
```

Comment:
Because the selector groups work based on `Combinator`s I wasn't sure whether we needed a change in how selectors are grouped, so instead I opted to have a boolean flag on a `block.nested` to determine whether the block is nested

> The [nesting selector](https://www.w3.org/TR/css-nesting-1/#nesting-selector) is allowed anywhere in a [compound selector](https://www.w3.org/TR/selectors-4/#compound), even before a [type selector](https://www.w3.org/TR/selectors-4/#type-selector), violating the normal restrictions on ordering within a compound selector.